nebula_hw_interfaces/src/nebula_hesai_hw_interfaces/hesai_hw_interface.cpp

```diff
@@ -2336,6 +2336,12 @@ Status HesaiHwInterface::SetPtpConfig(
   buf_vec.emplace_back((len >> 8) & 0xff);
   buf_vec.emplace_back((len >> 0) & 0xff);
 
+  if (sensor_configuration_->sensor_model == SensorModel::HESAI_PANDAR128_E4X) {
+    if (profile != static_cast<int>(PtpProfile::IEEE_802_1AS_AUTO)) {
+      return Status::SENSOR_CONFIG_ERROR;
+    }
+    profile = 3; // OT128 has a different definition of PTP profile
+  }
```

Comment:
I just realized, we also need to update the size in the header to 4; maybe move this above `int len = 6;` above and introduce a case `else if (profile == 3) { len = 4; }`

Reply:
Resolved in https://github.com/tier4/nebula/pull/125/commits/37d7b0f4a1a6ec2d5bffc2a375489dfaae253830